### PR TITLE
Updated to latest version

### PR DESCRIPTION
**Pull request by PP-Chris**
Merged master from parent repository. Fixes https://github.com/dtail-platform/pixelpool-unreal-gltf-runtime/issues/11

### CODE REVIEW

File: Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp [hunk 1]
```diff
@@ -16,6 +19,14 @@ AglTFRuntimeAssetActor::AglTFRuntimeAssetActor()
 	RootComponent = AssetRoot;
 	bAllowNodeAnimations = true;
 	bStaticMeshesAsSkeletal = false;
+	bAllowSkeletalAnimations = true;
```
Comment (RickPixelPool) [inline Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp:22]: Should we not allow these features until we confirm it works and/or we should support it? For example, bAllowLights could mess with how we want to light our products in PDP

File: Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp [hunk 2]
```diff
@@ -13,6 +14,9 @@ AglTFRuntimeAssetActorAsync::AglTFRuntimeAssetActorAsync()
 
 	AssetRoot = CreateDefaultSubobject<USceneComponent>(TEXT("AssetRoot"));
 	RootComponent = AssetRoot;
+
+	bShowWhileLoading = true;
```
Comment (RickPixelPool) [inline Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp:18]: Can we verify if this means loading looks differently now and if so it is acceptable

File: Source/glTFRuntime/Private/glTFRuntimeFunctionLibrary.cpp [hunk 3]
```diff
@@ -30,18 +36,97 @@ UglTFRuntimeAsset* UglTFRuntimeFunctionLibrary::glTFLoadAssetFromFilename(const
 	return Asset;
 }
 
+void UglTFRuntimeFunctionLibrary::glTFLoadAssetFromFilenameAsync(const FString& Filename, const bool bPathRelativeToContent, const FglTFRuntimeConfig& LoaderConfig, const FglTFRuntimeHttpResponse& Completed)
```
Comment (RickPixelPool) [inline Source/glTFRuntime/Private/glTFRuntimeFunctionLibrary.cpp:39]: Perhaps this allows us to simplify our async code we have in `FGltfSceneComponentLoader`

File: Source/glTFRuntime/Private/glTFRuntimeParserMaterials.cpp [hunk 4]
```diff
@@ -1,22 +1,41 @@
...
-UMaterialInterface* FglTFRuntimeParser::LoadMaterial_Internal(const int32 Index, const FString& MaterialName, TSharedRef<FJsonObject> JsonMaterialObject, const FglTFRuntimeMaterialsConfig& MaterialsConfig, const bool bUseVertexColors)
+UMaterialInterface* FglTFRuntimeParser::LoadMaterial_Internal(const int32 Index, const FString& MaterialName, TSharedRef<FJsonObject> JsonMaterialObject, const FglTFRuntimeMaterialsConfig& MaterialsConfig, const bool bUseVertexColors, UMaterialInterface* ForceBaseMaterial)
 {
 	SCOPED_NAMED_EVENT(FglTFRuntimeParser_LoadMaterial_Internal, FColor::Magenta);
 	FglTFRuntimeMaterial RuntimeMaterial;
 
-	RuntimeMaterial.BaseSpecularFactor = MaterialsConfig.SpecularFactor;
+	const FString Generator = GetGenerator();
+	bool bSpecularAutoDetected = false;
+	if (Generator.Contains("Blender") || Generator.Contains("Unreal Engine"))
```
Comment (RickPixelPool) [inline Source/glTFRuntime/Private/glTFRuntimeParserMaterials.cpp:29]: We should check/verify why this change was added and that it looks correct

File: Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp (hunk 1 above)
Comment (RickPixelPool) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp:22]: Will follow up on this with Jorrit, we can leave as is for now

Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp:22]: I disabled them to be sure anyway

File: Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp (hunk 2 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp:18]: It doesn't do anything right now as we don't show anything until we've finished parsing

File: Source/glTFRuntime/Private/glTFRuntimeFunctionLibrary.cpp (hunk 3 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeFunctionLibrary.cpp:39]: Correct. I did a brief test with this but then I reverted. Do it again properly?

File: Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp (hunk 2 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp:18]: The bra, for example, shows grey, and then switches to pink after a few seconds. Acceptable?

File: Source/glTFRuntime/Private/glTFRuntimeFunctionLibrary.cpp (hunk 3 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeFunctionLibrary.cpp:39]: Never mind, because of the way that the callback is set up, we're better off sticking with our method

File: Source/glTFRuntime/Private/glTFRuntimeParserMaterials.cpp (hunk 4 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeParserMaterials.cpp:29]: If not made it UE or Blender, it defaults to what was previously set before the change. Specular value set to 0.5 otherwise. Looks okay to me.

File: Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp (hunk 1 above)
Comment (RickPixelPool) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp:22]: Talked to Jorrit and for now we should keep them as default and not disable. He will let us know later on if certain new features work out of the box or if we can start supporting it, but no need to disable it at this point.

File: Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp (hunk 2 above)
Comment (RickPixelPool) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp:18]: Can you make a recording and add it to the jira task so we can discuss if it is acceptable?

File: Source/glTFRuntime/Private/glTFRuntimeParserMaterials.cpp (hunk 4 above)
Comment (RickPixelPool) [reply Source/glTFRuntime/Private/glTFRuntimeParserMaterials.cpp:29]: Also notified Jorrit he is aware of this. Once there is a library of different glb cases we can cross-reference it looks correct between frontend-Spaces, but we can keep as is for now

File: Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp [hunk 5]
```diff
@@ -598,34 +588,6 @@ UStaticMesh* FglTFRuntimeParser::LoadStaticMesh_Internal(TSharedRef<FglTFRuntime
 
 		}
 #endif
-
-		if (StaticMeshConfig.bBuildLumenCards)
```
Comment (RickPixelPool) [inline Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp]: Should we not keep this code in? Was there a reason it causes an issue for our PDP viewer?

File: Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp [hunk 6]
```diff
@@ -228,79 +228,69 @@ UStaticMesh* FglTFRuntimeParser::LoadStaticMesh_Internal(TSharedRef<FglTFRuntime
 			LODIndices.AddUninitialized(NumVertexInstancesPerSection);
 
 			// Geometry generation
-			ParallelFor(NumVertexInstancesPerSection, [&](const int32 VertexInstanceSectionIndex)
-				{
-					uint32 VertexIndex = Primitive.Indices[VertexInstanceSectionIndex];
-					LODIndices[VertexInstanceBaseIndex + VertexInstanceSectionIndex] = VertexInstanceBaseIndex + VertexInstanceSectionIndex;
+			for (int32 VertexInstanceSectionIndex = 0; VertexInstanceSectionIndex < NumVertexInstancesPerSection; VertexInstanceSectionIndex++)
```
Comment (RickPixelPool) [inline Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp:232]: Can you add a comment block `// PIXELPOOL - [explanation]` and maybe even a reference link to explain why we apply this difference compared to the gltf plugin? That way we can easily come back to this later to see if our custom change can be removed

File: Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp (hunk 5 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp]: Re-added and it's working fine

File: Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp (hunk 2 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActorAsync.cpp:18]: I think we agreed that this was okay

File: Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp (hunk 5 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp]: I think we agreed that this was okay

File: Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp (hunk 6 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp:232]: Good idea. Done

File: Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp (hunk 1 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeAssetActor.cpp:22]: Disregard my previous comment. It turns out that I didn't push the changes. Weird

File: Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp (hunk 5 above)
Comment (PP-Chris) [reply Source/glTFRuntime/Private/glTFRuntimeParserStaticMeshes.cpp]: Back in and working
